File: Sprint 2/Regimen.java
import java.util.*;
import java.time.*;

public class Regimen{
    private Patient patient;
    private Doctor doctor;
    private String medication;
    private String direction;
    private List<Record> records;
    
    
    Regimen(Patient patient, Doctor doctor, String medication, String direction){
        this.patient = patient;
        this.doctor = doctor;
        this.medication = medication;
        this.direction = direction;
        this.records = new ArrayList<>();
    }

    public void createRecord(LocalDate date, Boolean adherence){
        Record record = new Record(date, adherence);
        addRecord(record);
    }

    public void addRecord(Record record){
        records.add(record);
    }

    public Patient getPatient() {
        return patient;
    }

    public Doctor getDoctor() {
        return doctor;
    }

    public String getMedication() {
        return medication;
    }

    public String getDirection() {
        return direction;
    }

    public List<Record> getRecords() {
        return records;
    }

    public Record getRecord(int i) {
        return records.get(i);
    }
}

File: Sprint 3/Regimen.js
import { Record } from './Record.js';

export class Regimen {
    constructor(patient, doctor, medication, direction) {
        this.patient = patient;
        this.doctor = doctor;
        this.medication = medication;
        this.direction = direction;
        this.records = [];
    }

    createRecord(date, adherence) {
        const record = new Record(date, adherence);
        this.addRecord(record);
    }

    addRecord(record) {
        this.records.push(record);
    }

    getPatient() {
        return this.patient;
    }

    getDoctor() {
        return this.doctor;
    }

    getMedication() {
        return this.medication;
    }

    getDirection() {
        return this.direction;
    }

    getRecords() {
        return this.records;
    }

    getRecord(i) {
        return this.records[i];
    }
}